collect_app/src/main/java/org/odk/collect/android/formhierarchy/FormHierarchyFragment.java

```diff
@@ -106,7 +111,13 @@ public void onAttach(@NonNull Context context) {
         requireActivity().setTitle(formEntryViewModel.getFormController().getFormTitle());
         startIndex = formEntryViewModel.getFormController().getFormIndex();
 
-        MaterialProgressDialogFragment.showOn(this, formHierarchyViewModel.isCloning(), getParentFragmentManager(), () -> {
+        formHierarchyViewModel.getInstanceEditResult().observe(this, instanceEditResult -> {
```

Comment:
I think there's still going to be a problem here if the user navigates away from Collect and back again while cloning and the Activity gets recreated (clearing view model state). I'm going to test that and update https://github.com/getodk/collect/issues/6691 though, as I think the new cases protecting old edits will handle a lot of this.